docs/source/cuda/examples.rst

```diff
@@ -3,6 +3,397 @@
+it available for use inside CUDA kernels. This can be very useful for users that are migrating workflows from CPU to GPU as 
+they can directly reuse potential business logic with fewer code changes.
+
+Take the following example function:
+
+.. literalinclude:: ../../../numba/cuda/tests/doc_examples/test_cpu_gpu_compat.py
+   :language: python
+   :caption: from ``test_ex_cpu_gpu_compat`` in ``numba/cuda/tests/doc_examples/test_cpu_gpu_compat.py``
+   :start-after: ex_cpu_gpu_compat.define.begin
+   :end-before: ex_cpu_gpu_compat.define.end
+   :dedent: 8
+   :linenos:
+
+The function ``business_logic`` can be run standalone in compiled form on the CPU:
+
+.. literalinclude:: ../../../numba/cuda/tests/doc_examples/test_cpu_gpu_compat.py
+   :language: python
+   :caption: from ``test_ex_cpu_gpu_compat`` in ``numba/cuda/tests/doc_examples/test_cpu_gpu_compat.py``
+   :start-after: ex_cpu_gpu_compat.cpurun.begin
+   :end-before: ex_cpu_gpu_compat.cpurun.end
+   :dedent: 8
+   :linenos:
+
+It can also be directly reused threadwise inside a GPU kernel. For example one may 
+generate some vectors to represent ``x``, ``y``, and ``z``:
+
+.. literalinclude:: ../../../numba/cuda/tests/doc_examples/test_cpu_gpu_compat.py
+   :language: python
+   :caption: from ``test_ex_cpu_gpu_compat`` in ``numba/cuda/tests/doc_examples/test_cpu_gpu_compat.py``
+   :start-after: ex_cpu_gpu_compat.allocate.begin
+   :end-before: ex_cpu_gpu_compat.allocate.end
+   :dedent: 8
+   :linenos:
+
+And a numba kernel referencing the decorated function:
+
+.. literalinclude:: ../../../numba/cuda/tests/doc_examples/test_cpu_gpu_compat.py
+   :language: python
+   :caption: from ``test_ex_cpu_gpu_compat`` in ``numba/cuda/tests/doc_examples/test_cpu_gpu_compat.py``
+   :start-after: ex_cpu_gpu_compat.usegpu.begin
+   :end-before: ex_cpu_gpu_compat.usegpu.end
+   :dedent: 8
+   :linenos:
+
+This kernel can be invoked in the normal way:
+
+.. literalinclude:: ../../../numba/cuda/tests/doc_examples/test_cpu_gpu_compat.py
+   :language: python
+   :caption: from ``test_ex_cpu_gpu_compat`` in ``numba/cuda/tests/doc_examples/test_cpu_gpu_compat.py``
+   :start-after: ex_cpu_gpu_compat.launch.begin
+   :end-before: ex_cpu_gpu_compat.launch.end
+   :dedent: 8
+   :linenos:
+
+.. _cuda_montecarlo:
+
+Monte Carlo Integration
+=======================
+
+This example shows how to use Numba to approximate the value of a definite integral by rapidly generating 
+random numbers on the GPU. A detailed description of the mathematical mechanics of Monte Carlo integeration 
+is out of the scope of the example, but it can briefly be described as an averaging process where the area 
+under the curve is approximated by taking the average of many rectangles formed by its function values.
+
+In addition, this example shows how to perform reductions in numba using the :func:`numba.cuda.reduce` API.
```

Comment:
The link doesn't work because `reduce` is an alias, I think:

```suggestion
In addition, this example shows how to perform reductions in numba using the 
:func:`cuda.reduce() <numba.cuda.Reduce>` API.
```